test/support/generator/deal-investor-generator.ts

```diff
@@ -0,0 +1,56 @@
+import { DealInvestor } from '@ukef/modules/deal-investor/deal-investor.interface';
+
+import { AcbsGetDealPartyResponseDto } from '../../../src/modules/acbs/dto/acbs-get-deal-party-response.dto';
+import { AbstractGenerator } from './abstract-generator';
+
+export class DealInvestorGenerator extends AbstractGenerator<DealInvestor, GenerateResult, GenerateOptions> {
+  protected generateValues(): DealInvestor {
+    return {
+      dealIdentifier: this.valueGenerator.ukefId(),
+      portfolioIdentifier: this.valueGenerator.string(),
+      lenderType: { LenderTypeCode: this.valueGenerator.string() },
+      effectiveDate: this.valueGenerator.dateString(),
+      expiryDate: this.valueGenerator.dateString(),
+      isExpiryDateMaximum: this.valueGenerator.boolean(),
+      maximumLiability: this.valueGenerator.nonnegativeFloat(2),
+    };
+  }
+
+  protected transformRawValuesToGeneratedValues(values: DealInvestor[], { dealIdentifier, portfolioIdentifier }: GenerateOptions): GenerateResult {
+    const dealInvestorsInAcbs: AcbsGetDealPartyResponseDto[] = values.map((v) => ({
+      EffectiveDate: v.effectiveDate,
+      ExpirationDate: v.expiryDate,
+      IsExpirationDateMaximum: v.isExpiryDateMaximum,
+      LenderType: { LenderTypeCode: v.lenderType.LenderTypeCode },
+      LimitAmount: v.maximumLiability,
+    }));
+
+    const dealInvestors = values.map((v) => ({
+      dealIdentifier: dealIdentifier,
+      portfolioIdentifier: portfolioIdentifier,
+      lenderType: { LenderTypeCode: v.lenderType.LenderTypeCode },
+      effectiveDate: v.effectiveDate,
+      expiryDate: v.expiryDate,
+      isExpiryDateMaximum: v.isExpiryDateMaximum,
+      maximumLiability: v.maximumLiability,
+    }));
+
+    const dealInvestorsFromApi = dealInvestors;
+
+    return {
+      dealInvestorsInAcbs,
+      dealInvestors,
```

Comment:
Would it be clearer if `dealInvestors` here was renamed `dealInvestorsFromService`?

Reply:
Renamed and left 2 returned values.